src/generator.cpp

```diff
@@ -1682,8 +1682,8 @@ UnitsMap processEquationUnitsAst(const GeneratorEquationAstPtr &ast, UnitsMap un
                     VariablePtr variable = getVariable(ast);
                     ComponentPtr component = (variable != nullptr) ? std::dynamic_pointer_cast<Component>(variable->parent()) : nullptr;
                     ModelPtr model = (component != nullptr) ? owningModel(component) : nullptr;
-                    std::string compName = (component != nullptr) ? component->name() : nullptr;
-                    std::string modelName = (model != nullptr) ? model->name() : nullptr;
+                    std::string compName = (component != nullptr) ? component->name() : " ";
```

Comment:
Can't help but wonder why `" "` rather than `""`? (Keeping in mind that I have no idea what the code is supposed to be doing... :))

Reply:
No reason at all, actually. I'm just committing to make sure unit tests pass but coverage certainly won't - I'll change it once I'm satisfied with the test I have (which is a work in progress, I got rid of the old ones).